File: 0083_Remove Duplicates from Sorted List/RemoveDuplicatesfromSortedList.cs

/**
* Definition for singly-linked list.
* public class ListNode {
*     public int val;
*     public ListNode next;
*     public ListNode(int x) { val = x; }
* }
*/
public class Solution {
   public ListNode DeleteDuplicates(ListNode head) {
       if(head == null || head.next == null) return head;
       var current = head;
       while(current!=null)
       {
           var next = current.next;
           while(next!=null && next.val == current.val)
           {
               current.next = next.next;
               next = current.next;
           }

           current = next;
       }

       return head;      
   }
}

File: 0083_Remove Duplicates from Sorted List/RemoveDuplicatesfromSortedList.java
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode p = head;
        
        while(p!=null) {
            while(p.next!=null && p.val == p.next.val)
                p.next = p.next.next;
            p = p.next;
        }
        
        return head;
    }
}